Side Menu Component (v1)

mbhsiddiqui opened the pull request:
I've designed the initial side menu. There still some updates to this that I will push tomorrow. Currently, the side menu displays the mutable features but the changes cannot be applied (upcoming in later commit).

CCSCovenant committed:
prototype of tie/slur
adjust interface

CCSCovenant committed:
fix bug with slur/tied display

CCSCovenant committed:
fix bug with re-alignment

Commit:
Side Menu Component (v1)

Side Menu with drawer component for ease of access

CCSCovenant committed:
Merge remote-tracking branch 'origin/visualizer_refactor' into visualizer_refactor

CCSCovenant reviewed (changes requested):
![image](https://user-images.githubusercontent.com/23089796/161737250-c5d468d9-a508-41d7-a288-5a863fd9307e.png)
bug while changing the windows size

Commit:
Side Menu Bug Fix

Now supports window resize

CCSCovenant committed:
fix bug with re-alignment

CCSCovenant commented:
GUI and logic are not connected. please fix it
when a element is selected, setSElement(VElement vElement) in the Selected will be called.

get config list by vElement.getConfigAbleList();

refresh sidebar that pointer to that vElement setting
call vElement.updateConfig(); to update the settings
finally call visualizer.alignment() to apply change.

CCSCovenant committed:
update gui;
- add spinner to go to page
- add spinner to go to measure

CCSCovenant committed:
updater gradle

CCSCovenant committed:
support pdf export
fix bug that ExtensionFilter is not working

CCSCovenant committed:
support blend/
- support dot in the guitar
- fix problem with tie alignment

CCSCovenant committed:
fix alignment issue

CCSCovenant committed:
fix dot position

CCSCovenant committed:
update test for new visualizer
-optimize memory use

CCSCovenant committed:
fix problem with the subclass-super class relationship

CCSCovenant committed:
Merge branch 'visualizer_kuimou' into visualizer_refactor

# Conflicts:
#	src/main/resources/GUI/previewMXL.fxml

CCSCovenant committed:
Merge remote-tracking branch 'origin/visualizer_refactor' into visualizer_refactor

# Conflicts:
#	src/main/resources/GUI/previewMXL.fxml

CCSCovenant committed:
fix fxml error

CCSCovenant committed:
optimize gotoMeasure funtion

CCSCovenant committed:
add getNote function in the visualizer

CCSCovenant committed:
add a refresh icon
adjust slur/tied

Commit:
Updated previewMXL

- Adjusted play and export PDF buttons position
- Fixed bug for window resize for play and export PDF buttons

CCSCovenant committed:
add limit map

CCSCovenant committed:
Merge branch 'visualizer_kuimou' into visualizer_refactor

# Conflicts:
#	src/main/java/GUI/PreviewViewController.java
#	src/main/resources/GUI/previewMXL.fxml

CCSCovenant committed:
update all element's limit map
